spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/LocalizedQueueConnectionFactory.java

```diff
@@ -372,4 +343,113 @@ public void destroy() {
 		resetConnection();
 	}
 
+	/**
+	 * Used to obtain a connection factory for the queue leader.
+	 *
+	 * @param <T> the client type.
+	 * @since 2.4.8
```

Comment:
Why have you decided to back-port this fix?
Do you have an evidence that Spring Framework 5.3.x has moved to Commons Http Client 5.0?

Reply:
My plan is to back port but with the default `NodeLocator` being a `RabbitMQHopNodeLocator` (immediately deprecated) and make the `WebFluxNodeLocator` an optional component.

See the issue; the problem is HOP will be out of support before 2.4.x is, so we will need an option in 2.4.x that doesn't depend on HOP, in case of future REST changes in RabbitMQ.